File: trunk/server/object/status.cc
#include "status.h"

Status::Status(int status_id,
               const string& user_id,
               int problem_id,
               int contest_id,
               int time,
               int memory,
               const string& submit_time,
               int result,
               int language,
               int code_id,
               int code_length,
               const string& submit_ip,
               int error_id)
: status_id_(status_id),
  user_id_(user_id),
  problem_id_(problem_id),
  contest_id_(contest_id),
  time_(time),
  memory_(memory),
  submit_time_(submit_time),
  result_(result),
  language_(language),
  code_id_(code_id),
  code_length_(code_length),
  submit_ip_(submit_ip),
  error_id_(error_id){
 }
  
         
int Status::getStatusId() const{
  return status_id_;
}

string Status::getUserId() const{
  return user_id_;
}

int Status::getProblemId() const{
  return problem_id_;
}

int Status::getContestId() const{
  return contest_id_;
}

int Status::getTime() const{
  return time_;
}

int Status::getMemory() const{
  return memory_;
}

string Status::getSubmitTime() const{
  return submit_time_;
}

int Status::getResult() const{
  return result_;
}

int Status::getLanguage() const{
  return language_;
}

int Status::getCodeId() const{
  return code_id_;
}

int Status::getCodeLength() const{
  return code_length_;
}

string Status::getSubmitIp() const{
  return submit_ip_;
}

int Status::getErrorId() const{
  return error_id_;
}

  
void Status::setStatusId(int status_id){
  status_id_ = status_id;
}

void Status::setUseId(const string& user_id){
  user_id_ = user_id;
}

void Status::setProblemId(int problem_id){
  problem_id_ = problem_id;
}

void Status::setContestId(int contest_id){
  contest_id_ = contest_id;
}

void Status::setTime(int time){
  time_ = time;
}

void Status::setMemory(int memory){
  memory_ = memory;
}

void Status::setSubmitTime(const string& submit_time){
  submit_time_ = submit_time;
}

void Status::setResult(int result){
  result_ = result;
}

void Status::setLanguage(int language){
  language_ = language;
}

void Status::setCodeId(int code_id){
  code_id_ = code_id;
}

void Status::setCodeLength(int code_length){
  code_length_ = code_length;
}

void Status::setSubmitIp(const string& submit_ip){
  submit_ip_ = submit_ip;
}

void Status::setErrorId(int error_id){
  error_id_ = error_id;
}

File: client/cn/edu/whu/flood/object/Status.java
package cn.edu.whu.flood.object;

public class Status {
    
  private int statusId;
  private int problemId;
  private int contestId;
  private int result;
  private int memory;
  private int time;
  private int language;
  private int codeLength;
  private String submitTime;
  private boolean shareCode;
  
  public boolean isShareCode() {
	return shareCode;
  }


  public void setShareCode(boolean shareCode) {
	this.shareCode = shareCode;
  }


  public Status() {
      
  }
  

  public Status(int statusId, 
		        int problemId, 
		        int contestId, 
		        int result, 
		        int memory, 
		        int time, 
		        int language, 
		        int codeLength, 
		        String submitTime) {
	super();
	this.statusId = statusId;
	this.problemId = problemId;
	this.contestId = contestId;
	this.result = result;
	this.memory = memory;
	this.time = time;
	this.language = language;
	this.codeLength = codeLength;
	this.submitTime = submitTime;
  }

  public int getCodeLength() {
	return codeLength;
  }

  public void setCodeLength(int codeLength) {
	this.codeLength = codeLength;
  }

  public int getContestId() {
	return contestId;
  }

  public void setContestId(int contestId) {
	this.contestId = contestId;
  }

  public int getLanguage() {
	return language;
  }

  public void setLanguage(int language) {
	this.language = language;
  }

  public int getMemory() {
	return memory;
  }

  public void setMemory(int memory) {
	this.memory = memory;
  }

  public int getProblemId() {
	return problemId;
  }

  public void setProblemId(int problemId) {
	this.problemId = problemId;
  }

  public int getResult() {
	return result;
  }

  public void setResult(int result) {
	this.result = result;
  }

  public int getStatusId() {
	return statusId;
  }

  public void setStatusId(int statusId) {
	this.statusId = statusId;
  }

  public String getSubmitTime() {
	return submitTime;
  }

  public void setSubmitTime(String submitTime) {
	this.submitTime = submitTime;
  }

  public int getTime() {
	return time;
  }

  public void setTime(int time) {
	this.time = time;
  }
}
